Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py

```diff
@@ -814,24 +899,46 @@ def main():  # pragma: no cover
     """
         PARSE AND VALIDATE INTEGRATION PARAMS
     """
-    command = demisto.command()
-    params = demisto.params()
+    params: dict = demisto.params()
+    app_id = params.get('app_id')
+    tenant_id = params.get('tenant_id')
+    auth_mode = params.get('auth_mode', 'legacy')
+    enc_key = (params.get('credentials') or {}).get('password')
+    private_key = params.get('private_key')
+
+    verify = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    certificate_thumbprint = params.get('certificate_thumbprint')
+
     token = params.get('token')
     base_url = f'{params.get("url")}/api/v1'
-    verify_certificate = not params.get('insecure', False)
     first_fetch = params.get('first_fetch')
     max_results = params.get('max_fetch')
-    proxy = params.get('proxy', False)
     severity = params.get('severity')
     resolution_status = params.get('resolution_status')
     look_back = arg_to_number(params.get('look_back')) or 0
-    LOG(f'Command being called is {command}')
+
+    command = demisto.command()
+    args = demisto.args()
+
     try:
+        if not app_id and not token:
+            raise Exception('Application ID must be provided.')
+
         client = Client(
+            app_id=app_id,
+            verify=verify,
             base_url=base_url,
-            verify=verify_certificate,
-            headers={'Authorization': f'Token {token}'},
-            proxy=proxy)
+            proxy=proxy,
+            tenant_id=tenant_id,
+            enc_key=enc_key,
+            auth_mode=auth_mode,
+            certificate_thumbprint=certificate_thumbprint,
+            private_key=private_key,
+            headers={'Authorization': f'Token {token}'}
+        )
+
+        LOG(f'Command being called is {command}')
 
         if command == 'test-module':
```

Comment:
I am not sure we can do test-module here, I think we'll have to check which authentication method is used and in case we have device code then instruct users to use the test connection command. PLease validate me together with @yucohen